strr-host-pm-web/app/components/summary/SupportingInfo.vue

```diff
@@ -52,9 +52,11 @@ const supportingInfo = computed(() => {
 })
 
 const handleUploadDocument = async (uiDoc: UiDocument, appRegNumber: string | number) => {
-  await isRegistration.value
-    ? docStore.addDocumentToRegistration(uiDoc, appRegNumber as number)
-    : docStore.addDocumentToApplication(uiDoc, appRegNumber as string)
+  if (isRegistration.value && (hasRegistrationNoc.value || needsBusinessLicenseDocumentUpload.value)) {
```

Comment:
So for Registration without NOC or BL Doc Upload, we would Add Documents to Application?